javascript/ql/src/semmle/javascript/security/dataflow/Xss.qll

```diff
@@ -34,7 +34,14 @@ module Shared {
   class MetacharEscapeSanitizer extends Sanitizer, StringReplaceCall {
     MetacharEscapeSanitizer() {
       isGlobal() and
-      RegExp::alwaysMatchesMetaCharacter(getRegExp().getRoot(), ["<", "'", "\""])
+      (
+        RegExp::alwaysMatchesMetaCharacter(getRegExp().getRoot(), ["<", "'", "\""])
+        or
+        // or it's a global inverted char class.
+        getRegExp().getRoot().(RegExpCharacterClass).isInverted()
```

Comment:
Should we check which `RegExpCharacterClass` we are dealing with?

Reply:
Hmmm. Possibly yes.  

I found the `RegExp::isWildcardLike` predicate, which I think matches nicely, so I refactored to use that predicate.